arbitrum-docs/stylus/stylus-quickstart.md

```diff
@@ -11,209 +11,242 @@ import PublicPreviewBannerPartial from '../partials/_public-preview-banner-parti
 
-To check whether or not your program will successfully **deploy and activate** onchain, use the `cargo stylus check` subcommand:
+## Checking if your Stylus project is valid
 
-```
+By running the ' cargo stylus check ' command, you can check if your program can be successfully **deployed and activated** onchain.
+
+**Important:** Ensure you run Docker for this command to work correctly.
+
+```shell
 cargo stylus check
 ```
 
-This command will attempt to verify that your program can be deployed and activated onchain without requiring a transaction by specifying a JSON-RPC endpoint. See `cargo stylus check --help` for more options.
+`cargo stylus check` executes a dry-run on your project by compiling your contract to WASM and verifying if it can be deployed and activated onchain.
 
-If the command above fails, you'll see detailed information about why your WASM will be rejected:
 
-```
+If the command above fails, you'll see detailed information about why your contract would be rejected:
+
+```shell
 Reading WASM file at bad-export.wat
 Compressed WASM size: 55 B
-Stylus checks failed: program predeployment check failed when checking against
+Stylus checks failed: program pre-deployment check failed when checking against
 ARB_WASM_ADDRESS 0x0000…0071: (code: -32000, message: program activation failed: failed to parse program)
 
 Caused by:
     binary exports reserved symbol stylus_ink_left
 
 Location:
     prover/src/binary.rs:493:9, data: None)
-
 ```
 
-To read more about what counts as valid vs. invalid user WASM programs, see [VALID_WASM](https://github.com/OffchainLabs/cargo-stylus/blob/main/main/VALID_WASM.md). If your program succeeds, you'll see the following message:
+When a program fails the check, it means that there was a problem with the WASM binary generated by `cargo-stylus`.
+To read more about what counts as valid vs. invalid user WASM contracts, you can visit the [Invalid Stylus WASM Contracts explainer](https://github.com/OffchainLabs/cargo-stylus/blob/main/main/VALID_WASM.md).
 
-```
+If your program succeeds, you'll see something like this:
+
+```shell
 Finished release [optimized] target(s) in 1.88s
 Reading WASM file at hello-stylus/target/wasm32-unknown-unknown/release/hello-stylus.wasm
 Compressed WASM size: 3 KB
 Program succeeded Stylus onchain activation checks with Stylus version: 1
 ```
 
-Note that running `cargo stylus check` may take a few minutes, especially on a fresh build. If you're running the check or deploy commands, a message at the start will inform you that this process might take some time. For faster checks, you can use the `--no-verify` option. Also, please ensure that Docker is up and running, as these commands are required to be executed properly.
+Note that running `cargo stylus check` may take a few minutes, especially if you're checking a contract for the first time. 
+You can use the `-no-verify` option for faster checks.
```

Comment:
```suggestion
You can use the `--no-verify` option for faster checks. Keep in mind that this will use your local architecture instead of a docker container to run `cargo-stylus`.
```

Reply:
In v0.5.0 `cargo stylus check` runs outside of docker by default so `--no-verify` isn't relevant there anymore. It'll still be relevant for `deploy` or `verify`

Reply:
I believe it still runs docker (just checked on 0.5.3), although it doesn't show the full message that `deploy` or `verify` do. But, now that I think about it, I'm not sure if mentioning `--no-verify` adds a lot of value in the quickstart 🤔 .

Reply:
I removed  every reference to`--no-verify`